multiversx_sdk/network_providers/http_resources.py

```diff
@@ -0,0 +1,33 @@
+import base64
+from typing import Any, Dict
+
+from multiversx_sdk.core.smart_contract_query import (
+    SmartContractQuery, SmartContractQueryResponse)
+
+
+def smart_contract_query_to_dictionary(query: SmartContractQuery) -> Dict[str, Any]:
+    request: Dict[str, Any] = {
+        'scAddress': query.contract,
+        'funcName': query.function,
+        'value': str(query.value if query.value else 0),
+        'args': [arg.hex() for arg in query.arguments]
+    }
+    caller = query.caller
```

Comment:
Can be moved under `if`.

Reply:
Optionally, let's think of a naming convention for functions in this file. E.g.:

- `smart_contract_query_to_vm_values_query_request` 
- `vm_values_query_response_to_smart_contract_query_response`

(just an idea).

Reply:
Removed the `caller` variable as it was unnecessary. Also renamed the functions.